tests/unit/commands/publish/test_command.py

```diff
@@ -0,0 +1,147 @@
+"""Test sam publish CLI."""
+import json
+from unittest import TestCase
+from mock import patch, call, Mock
+
+from botocore.exceptions import ClientError
+
+from serverlessrepo.exceptions import ServerlessRepoError
+from serverlessrepo.publish import CREATE_APPLICATION, UPDATE_APPLICATION
+
+from samcli.commands.publish.command import do_cli as publish_cli
+from samcli.commands.local.cli_common.user_exceptions import SamTemplateNotFoundException
+from samcli.commands.exceptions import UserException
+
+
+class TestCli(TestCase):
+
+    def setUp(self):
+        self.template = "./template"
+        self.application_id = "arn:aws:serverlessrepo:us-east-1:123456789012:applications/hello"
+        self.ctx_mock = Mock()
+        self.ctx_mock.region = "us-east-1"
+        self.console_link = "Click the link below to view your application in AWS console:\n" \
+            "https://console.aws.amazon.com/serverlessrepo/home?region={}#/published-applications/{}"
+
+        path_patcher = patch('samcli.commands.publish.command.os.path')
+        self.path_mock = path_patcher.start()
+        self.path_mock.exists.return_value = True
```

Comment:
If this changes between tests, don't set the value in the `setUp`. It makes it more difficult to understand each test if the variables aren't assigned consistently.